gson/src/main/java/com/google/gson/internal/Streams.java

```diff
@@ -103,8 +103,15 @@ private static final class AppendableWriter extends Writer {
     /**
      * A mutable char sequence pointing at a single char[].
      */
-    static class CurrentWrite implements CharSequence {
-      char[] chars;
+    private static class CurrentWrite implements CharSequence {
+      private char[] chars;
+      private String cachedString;
+
+      public void setChars(char[] chars) {
```

Comment:
I don't think we need `public` here, do we?

Reply:
You are right, package-private or `private` would work as well. My intention was to indicate that this is the proper way to update the `chars` value, but that is not so important. I think for Gson users it would not matter either way because the class is `private` (and in the `internal` package).